libraries/botframework-schema/etc/botframework-schema.api.md

```diff
@@ -953,11 +953,47 @@ export interface MeetingInfo {
     organizer: TeamsChannelAccount;
 }
 
+// @public
+export type MeetingNotification = TargetedMeetingNotification;
+
+// @public
+export interface MeetingNotificationBase<T> {
+    type: string;
+    value: T;
+}
+
+// @public
+export interface MeetingNotificationChannelData {
+    onBehalfOf?: OnBehalfOf[];
+}
+
+// @public
+export interface MeetingNotificationRecipientFailureInfo {
+    errorCode: string;
+    failureReason: string;
+    recipientMri: string;
+}
+
+// @public
+export interface MeetingNotificationResponse {
+    recipientsFailureInfo?: MeetingNotificationRecipientFailureInfo[];
+}
+
+// @public (undocumented)
+export interface MeetingStageSurface<T> {
```

Comment:
Does it make sense to add _extends Surface_ for` interface MeetingStageSurface<T>`
_export interface MeetingStageSurface<T> extends Surface {
     ...
}_

Reply:
MeetingStageSurface<T> extends Surface, simply copies the properties/type pair in Surface to MeetingStageSurface<T>, nothing more. So then there's not much reason to have a base interface since it will have only one child interface.

Reply:
But aren't we going to support more surfaces in the future? Such as TabSurface or ContentBubbleSurface?

Reply:
Let's say we have a `Surface` interface with `surface` property with type string. In the sub-interface (think subclass) `MeetingStageSurface<T>` we will be overriding it to something like `surface: "meetingStageSurface"`. Now if we assign a variable the type `Surface`, it will not accept an object that has type `MeetingStageSurface<T>` even though it is a sub-interface of `Surface`. That is, polymorphism doesn't work. Now if the sub-interface did not extend the parent interface, and everything else remains the same, then behavior will be the exact same. This means that having `MeetingStageSurface<T>` extend  `Surface` being completely redundant.

Now in the future if we have `TabSurface`. We simply update `type MeetingSurface = MeetingStageSurface<T> | TabSurface` and any object of type `MeetingSurface` can be either of the two assigned types. This is pretty much the polymorphism equivalent in typescript.

Reply:
@singhk97 , I think @yingduyingdu's point is to check whether the current design aligns with what the model class hierarchy that we have discussed. 

If I am understanding your point correctly, though the implementation details might be different, this is the right way to (a) show the correct intention to developers and (b) support the future extensibility. Am I aligned with you? 

My assumption was for typescript, how it represents the class hierarchy would be very similar to the other existing OOP languages, so the current implementations gave me some surprise :) But my assumption might have been wrong :)

Reply:
Hey @singhk97 , I want to also have a conversation on this as well. 
First, we decided to make SDK changes on top of the existing one to show the correct intentions of the API to developers. One of the intentions is to let developers know that they can send any list of surfaces in one API.  

I don't think the current implementation shows that intention of the API. Also,I am not sure how the current implementation will support this sceanrio: when a developer sends both `MeetingStageSurface<T>` and `TabSurface` in one API request (and both surface will have completely different schemas), how would that work when there is no common class/interface between them?